src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -354,49 +357,55 @@ private static bool TryParseStatusCode(string responseHeader, int statusCodeStar
             long timeoutInMilliseconds,
             IntPtr context)
         {
-            // TODO: Handle <=0 values for the timeout
-            Task.Delay((int)timeoutInMilliseconds).ContinueWithStandard(context, (t, c) => CurlTimerElapsed((IntPtr)c));
+            if (timeoutInMilliseconds <= 0)
```

Comment:
Sorry.. I had meant to replace the TODO with a more detailed TODO. The only fix I was trying to put was passing negative value to Task.Delay. I still need to understand the documentation correctly and revisit the sample code. I think I have misinterpreted the "non-repeatable" part mentioned in the doc.